akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardingQueries.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding
+
+import scala.reflect.ClassTag
+
+import akka.annotation.InternalApi
+
+/** INTERNAL API */
+@InternalApi
+private[sharding] object ShardingQueries {
+
+  /**
+   * INTERNAL API
+   * The result of a group query and metadata.
+   *
+   * @param timedout the number of non-responses within the configured timeout, which
+   *                     could be indicative of several states, e.g. still in initialization,
+   *                     restart, heavily loaded and busy, where returning a simple zero is
+   *                     not indicative of the reason
+   * @param responses the number of responses received from the query
+   * @param total the total number of shards tracked versus a possible subset
+   * @param queried the number of shards queried, which could equal the total or be a
+   *                subset if this was a retry of those that timed out
+   * @tparam A
+   * @tparam B
+   */
+  @InternalApi
+  private[sharding] final case class ShardsQueryResult[A, B](
+      timedout: Seq[A],
+      responses: Seq[B],
+      total: Int,
+      queried: Int) {
```

Comment:
(it's not obvious to me that 'total' and 'queried' are useful, but as this is an internal message it's easy to refactor later)

Reply:
the total can be all shards whereas queried is either all of a subset we want to retry the failures only or do some other grouping that isn't all of them.